lib/fluent/plugin/in_tail.rb

```diff
@@ -201,7 +204,15 @@ def start
       end
 
       refresh_watchers unless @skip_refresh_on_startup
-      timer_execute(:in_tail_refresh_watchers, @refresh_interval, &method(:refresh_watchers))
+
+      @threads['in_tail_refresh_watchers'] = Thread.new(@refresh_interval) do |refresh_interval|
```

Comment:
can you use https://docs.fluentd.org/plugin-helper-overview/api-plugin-helper-thread